.github/workflows/mypy-clinic.yml

```diff
@@ -0,0 +1,32 @@
+name: Run mypy on Tools/clinic
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "Tools/clinic/**/*"
+      - ".github/workflows/mypy-clinic.yml"
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: pip
+          cache-dependency-path: .github/workflows/mypy-clinic.yml
+      - run: pip install mypy==1.3.0
```

Comment:
It would be nice if this dependency pin would be picked up by dependabot. I don't _think_ that's possible if we just stick it in the `.yml` workflow file, but maybe @hugovk can correct me if I'm wrong there!

One way of getting dependabot to update the dependency pin might be if we put the pin in a `dev-requirements.txt` file in `Tools/clinic/`. But that almost feels a bit silly if we only have one dependency.

Reply:
Correct, dependabot won't update pip dependencies in workflow files. (It will update GitHub Actions versions.)

You can specify a directory:

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#pip-and-pip-compile

For example, one project has https://github.com/jazzband/django-simple-history/blob/e368d839c981c60376cbffcdaba245b89766bff0/.github/dependabot.yml:

```yml
version: 2
updates:
  - package-ecosystem: "pip"
    directory: "/requirements"
    schedule:
      interval: "daily"
```

And has a load of `.txt` files in [`/requirements`](https://github.com/jazzband/django-simple-history/tree/e368d839c981c60376cbffcdaba245b89766bff0/requirements) which dependabot has been updating.

(`interval: "daily"` is too often, `"monthly"` is usually fine.)

Reply:
No, it is not possible. And as far as I know, many people here do not like dependabot PRs :(

Reply:
I think a once-a-month PR updating a mypy pin should be fine, personally. We already get dependabot PRs for GitHub Actions pins.